azurerm/internal/services/eventgrid/eventgrid_topic_resource.go

```diff
@@ -196,3 +301,134 @@ func resourceArmEventGridTopicDelete(d *schema.ResourceData, meta interface{}) e
 
 	return nil
 }
+
+func expandAzureRmEventgridTopicInputMapping(d *schema.ResourceData) *eventgrid.JSONInputSchemaMapping {
+	imf, imfok := d.GetOk("input_mapping_fields")
+
+	imdv, imdvok := d.GetOk("input_mapping_default_values")
+
+	if !imfok && !imdvok {
+		return nil
+	}
+
+	jismp := eventgrid.JSONInputSchemaMappingProperties{}
+
+	if imfok {
+		mappings := imf.([]interface{})
+		mapping := mappings[0].(map[string]interface{})
+
+		if id := mapping["id"].(string); id != "" {
+			jismp.ID = &eventgrid.JSONField{SourceField: &id}
+		}
+
+		if eventTime := mapping["event_time"].(string); eventTime != "" {
+			jismp.EventTime = &eventgrid.JSONField{SourceField: &eventTime}
+		}
+
+		if topic := mapping["topic"].(string); topic != "" {
+			jismp.Topic = &eventgrid.JSONField{SourceField: &topic}
+		}
+
+		if dataVersion := mapping["data_version"].(string); dataVersion != "" {
+			jismp.DataVersion = &eventgrid.JSONFieldWithDefault{SourceField: &dataVersion}
+		}
+
+		if subject := mapping["subject"].(string); subject != "" {
+			jismp.Subject = &eventgrid.JSONFieldWithDefault{SourceField: &subject}
+		}
+
+		if eventType := mapping["event_type"].(string); eventType != "" {
+			jismp.EventType = &eventgrid.JSONFieldWithDefault{SourceField: &eventType}
+		}
+	}
+
+	if imdvok {
+		mappings := imdv.([]interface{})
+		mapping := mappings[0].(map[string]interface{})
```

Comment:
Similar to the above, this could panic and we should check it before casting

Reply:
fixed